### ISSUE TITLE
native async await

### ISSUE THREAD

**Issue by adnan-kamili** [yortus/asyncawait#38]
When async await feature lands in V8, won't it break your library as async and await will become keyword in javascript.

**Comment by yortus** [comment]
Hi @adnan-kamili,

It should not break the library itself, because you can import the `async` and `await` functions using whatever names you want.

Furthermore, neither `async` nor `await` are becoming keywords in the normal sense. `async` is not becoming a keyword at all ([MDN ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords)) - I suspect this was a pragmatic approach, because otherwise it would break numerous scripts that depend on the very popular [async library](https://github.com/caolan/async).

The `await` story is more complicated. It is a future reserved keyword, but only when it is found in module code, where a 'module' in this case means an ES6 module, and only at the top scope. That means a compliant JS runtime won't allow `import await from 'asyncawait'` in an ES6 module, however it will continue to be a valid identifier inside all function bodies, even in functions that appear inside ES6 modules.

So TL;DR at some future time when V8 and node.js support ES6 modules, you'll still be able to `import async from 'asyncawait'` but you'll have to use a different name for importing the `await` function.

By that time native `async`/`await` will be available so most projects will be able to transition to that anyway. Although this library may still be maintained since it can do a number of things native `async`/`await` cannot, such as being customisable, and being able to await at arbitrary stack depths (useful for function code, eg inside `filter`/`forEach`/`map`/`reduce` callbacks).

Does this answer your question?

**Comment by adnan-kamili** [comment]
Thanks for the reply. It does answer the question well
